### DISCUSSION TITLE
Am I just being dumb?

### DISCUSSION THREAD

**Discussion by AaronAyante** [Mydayyy/Valheim-ServerSideMap#60] [Q&A] [answered]
OK, so my wife and I are playing together, and I'm hosting a dedicated server. We are running a number of mods, including Epic Valheim, using R2 ModMan as the mod manager. 

In both the Client and Server startup logs, I see:

[Warning:   BepInEx] Skipping [ServerSideMap 1.3.10.0] because a newer version exists (ServerSideMap 1.3.10.0)

and then

[Info   :   BepInEx] Loading [ServerSideMap 1.3.10.0]
[Info   :ServerSideMap] Store1: False
[Info   :ServerSideMap] Store2: False

Now, I did copy the ServerSideMap.dll out of the Mydayyy-ServerSideMap directory within the BepinEx\Plugins directory and place a copy in the Plugins folder directly, just in case. 

Now, I do also see this in my client:

[Info   :ServerSideMap] ZNet awake
and
[Info   :ServerSideMap] Minimap awake

and on the server

[Info   :ServerSideMap] World .explored save path: ./DedicatedServer/worlds_local/Serenity
[Info   :ServerSideMap] loaded from existing explore file

and 

12/31/2023 23:44:58: Console: [Info   :ServerSideMap] Registered Server Events

The problem is... pins aren't being synced across our clients... and when my wife reconnected, a bunch of pins that she had made last night weren't on her map anymore either.

I see lines such as "Server received pin by client" and "client received pin by server" but... nothing seems to be happening? I seem to see SOME of her pins, but if she logs out and then back in, her own pins are gone again.

**Comment by Mydayyy** [comment]
Greetings,

just to confirm, did you enable the sharing of pins on all three clients via the config file? (Your client, your wifes client and the server)?

**Reply by AaronAyante** [reply]
Yep - I setup the config on my system and copied it over to the server folder, and verified the same is set up on hers. EnableMapShare is true, EnableMarkerShare is true. We've both tried the /convertpins command as well. I see a Serenity.mod.serversidemap.explored and a .beforehs file, both reading just about 5 Mb, in the worlds_local file alongside the world itself. 

As far as I can tell, it *should* be working, so yeah, I'm perplexed haha

**Reply by AaronAyante** [reply]
Ok, just to validate - when someone disconnects, I see the following:

[Info   :ServerSideMap] Map saved. Pin Count: 75
01/02/2024 09:34:20: Console: [Info   :ServerSideMap] Map saved. Pin Count: 75
[Info   :ServerSideMap] World .explored save path: ./DedicatedServer/worlds_local/Serenity
01/02/2024 09:34:20: Console: [Info   :ServerSideMap] World .explored save path: ./DedicatedServer/worlds_local/Serenity

So it seems like, for whatever reason, we simply aren't retrieving the mapfile? We are also running AzuMapDetails, but I don't believe that should interfere (since it simply adds detailed info in a range of 50 meters around the player)?

**Reply by Mydayyy** [reply]
Greetings,

I don't know about AzuMapDetails - since I do not play Valheim actively anymore, I just keep this mod up to date with the game. Do you have the possibility to test the mod isolated without any other mods and try to reproduce the issue? That would be the fastest way to rule out any conflicts. But if AzuMapDetails does touch the pin functionality it could interfere.

But generally, it looks like the pins are picked up by the server, and if the client is sending the pins to the server it also means that this functionality is enabled on the client side via the config file correctly. In that case there might be an error during the retrieval phase of the pins. Can you check the player.log for any errors or alternatively upload it (and make sure it does not contain any sensitive information)?

**Reply by AaronAyante** [reply, marked as answer]
I am... now super confused, as it seems like things are working now, including pins that I make and then she sees a few moments later.

I'm not sure why, as nothing changed other than a server restart but... I'll take it!
